modules/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/SeqexecEngine.scala

```diff
@@ -283,12 +283,15 @@ object SeqexecEngine {
 
   def apply(settings: Settings) = new SeqexecEngine(settings)
 
-  private def decodeTops(s: String): Map[String, String] = s.split("=|,").grouped(2).map { case Array(k, v) => k.trim -> v.trim }.toMap
+  private def decodeTops(s: String): Map[String, String] =
+    s.split("=|,").grouped(2).collect {
+      case Array(k, v) => k.trim -> v.trim
+    }.toMap
 
   private def initSmartGCal(smartGCalHost: String, smartGCalLocation: String): Task[Unit] = {
     // SmartGCal always talks to GS
     val peer = new Peer(smartGCalHost, 8443, Site.GS)
-    Task.delay(Paths.get(smartGCalLocation)) >>= {p => Task.delay(SmartGcal.initialize(peer, p))}
+    Task.delay(Paths.get(smartGCalLocation)) >>= {p => Task.delay(SmartGcal.initialize(peer, p)) *> Task.now(()) }
```

Comment:
Similarly, I think you can `.void` the result if you don't need it here?

Also, I'm curious why this isn't just a `map`.